README.md

```diff
@@ -0,0 +1,125 @@
+# Errors utils
+
+Custom errors utilities
+
+## Installation
+
+```sh
+npm install errors-utils
+```
+
+## Example usage
+
+```ts
+import { assertAs, createNamespaceError } from 'errors-utils'
+import { name, version } from '../package.json' // use metadata from package.json
+
+const LibError = createNamespaceError('LIB', { package: name, version })
```

Comment:
This approach is a little different than what I had been imagining as now ErrorClass definitions can be spread out throughout the codebase.  This makes it easier to create new ones ad-hoc, but makes it harder to form a dedicated accounting of all error codes in our system.  I had imagined there would be a single file with all the error definitions so it'd be really easy to see the full set of errors we use and could more easily document them

Reply:
Yes that would be possible if we were using a monorepo with all our libraries but as it's not the case I think it's safer to namespace these so we don't have to coordinate all the time about the error codes we use.
For example the DocID lib could use the `DCD` namespace and IDX the `IDX` one and that's the only thing we need to coordinate over, then each library is safe to use the codes they want (error code `DCD123` is different from `IDX123`).

Reply:
But isn't one of the advantages of creating a centralized error handling system that we can have a global namespace of error codes?  What if one of our packages wants to send an error over the network to another?  Having a global set of codes would make that much easier.  It also means that if you're a developer (or us reading customer logs) and see an error code in a log message you know what error it refers to without having to consult the package-specific list of codes, so long as you know that it came from *one* of the packages built by 3boxlabs

Reply:
That's what we'd get from the namespaces and metadata, no? When reading the logs, if we have a stack like this we can see the errors happening in each package:
```js
[
  {code: 'IDX34', message: 'Request failed', metadata: { version: '0.2.1' }},
  {code: 'CRP765', message: 'Invalid input', metadata: { package: 'http-client', version: '0.16.4', }},
  {code: 'DCD3', message: 'Invalid multibase encoding', metadata: { package: 'docid', version: '0.4.0', }},
]
```

I think we should also have a dedicated repository with markdown files (let's say `ceramicnetwork/error-codes`), such as:
- `CRP.md`
- `DCD.md`
- `IDX.md`

With contents such as (in `DCD.md`):

```md
# DocID namespace (DCD)

## DCD1

Missing version byte in payload.

**Versions**: 0.3.5 - 0.7.0 *deprecated*
**Resolution**: use `DocID.validate()` to check input without throwing an error

...

## DCD5

Invalid multibase argument.

**Versions**: 0.4.2 - current
**Resolution**: see multibase compatibility docs
```

This way we can have deterministic URLs for error codes, ex `https://github.com/ceramicnetwork/error-codes/blob/master/DCD.md#DCD5`